packages/react-native-reanimated/apple/reanimated/apple/REAModule.mm

```diff
@@ -279,38 +276,18 @@ - (void)sendEventWithName:(NSString *)eventName body:(id)body
 RCT_EXPORT_BLOCKING_SYNCHRONOUS_METHOD(installTurboModule)
 {
   WorkletsModule *workletsModule = [_moduleRegistry moduleForName:"WorkletsModule"];
-  if (_isBridgeless) {
-#ifdef RCT_NEW_ARCH_ENABLED
-    RCTCxxBridge *cxxBridge = (RCTCxxBridge *)self.bridge;
-    auto &rnRuntime = *(jsi::Runtime *)cxxBridge.runtime;
-    auto executorFunction = ([executor = _runtimeExecutor](std::function<void(jsi::Runtime & runtime)> &&callback) {
-      // Convert to Objective-C block so it can be captured properly.
-      __block auto callbackBlock = callback;
-
-      [executor execute:^(jsi::Runtime &runtime) {
-        callbackBlock(runtime);
-      }];
-    });
-    auto reanimatedModuleProxy = reanimated::createReanimatedModuleBridgeless(
-        self, _moduleRegistry, rnRuntime, workletsModule, executorFunction);
-    [self attachReactEventListener];
-    [self commonInit:reanimatedModuleProxy withRnRuntime:rnRuntime];
-#else
-    [NSException raise:@"Missing bridge" format:@"[Reanimated] Failed to obtain the bridge."];
-#endif // RCT_NEW_ARCH_ENABLED
-  } else {
-    facebook::jsi::Runtime *jsiRuntime = [self.bridge respondsToSelector:@selector(runtime)]
-        ? reinterpret_cast<facebook::jsi::Runtime *>(self.bridge.runtime)
-        : nullptr;
 
-    if (jsiRuntime) {
-      auto reanimatedModuleProxy =
-          reanimated::createReanimatedModule(self, self.bridge, self.bridge.jsCallInvoker, workletsModule);
-      jsi::Runtime &rnRuntime = *jsiRuntime;
+  auto jsCallInvoker = self.bridge.jsCallInvoker;
+  auto jsiRuntime = reinterpret_cast<facebook::jsi::Runtime *>(self.bridge.runtime);
+
+  assert(jsiRuntime != nullptr);
+
+  auto reanimatedModuleProxy =
+      reanimated::createReanimatedModule(self, self.bridge, jsCallInvoker, workletsModule, _isBridgeless);
+
+  jsi::Runtime &rnRuntime = *jsiRuntime;
+  [self commonInit:reanimatedModuleProxy withRnRuntime:rnRuntime];
```

Comment:
We shouldn't use `self.bridge.jsCallInvoker` in bridgeless mode. You have to do the following:
```objc
#ifdef RCT_NEW_ARCH_ENABLED
@synthesize callInvoker = _callInvoker;
#endif // RCT_NEW_ARCH_ENABLED

auto jsCallInvoker = _callInvoker.callInvoker;
```

Let's restore the original implementation without refactoring the whole function. See https://github.com/software-mansion/react-native-reanimated/pull/6768/files#diff-06f7028a5f8a14ac525b70098bbf48937abe91707aea813bc4528ec9c166f5bc

Reply:
[ff3eeb2](https://github.com/software-mansion/react-native-reanimated/pull/6783/commits/ff3eeb2ec53004767ab8aa97c08813fc12fec41d)